src/vs/vscode.d.ts

```diff
@@ -1119,6 +1119,21 @@ declare module 'vscode' {
 		 */
 		edit(callback: (editBuilder: TextEditorEdit) => void, options?: { undoStopBefore: boolean; undoStopAfter: boolean; }): Thenable<boolean>;
 
+		/**
+		 * Change document language by Id
+		 *
+		 * @param languageId new language identifier
+		 */
+		setLanguageById(languageId: string): void;
+
+		/**
+		 * Change document language by Name
+		 *
+		 * @param languageName new language name
+		 */
+		setLanguageByName(languageName: string): void;
+
```

Comment:
We don't want those because they will change that editor instance (changing the mode changes the document instance and that changes the editor instance). Please remove these functions and what it takes to implement them